actionpack/lib/action_dispatch/routing/route_set.rb

```diff
@@ -261,6 +261,7 @@ def initialize(route, options, route_name)
 
           def call(t, method_name, args, inner_options, url_strategy)
             controller_options = t.url_options
+            controller_options[:_recall]&.delete(:id)
```

Comment:
Probably not an ideal solution, but this is the most minimal patch to get the test passing.